core/src/main/java/com/linecorp/armeria/server/docs/DocServiceBuilderImpl.java

```diff
@@ -0,0 +1,476 @@
+/*
+ *  Copyright 2017 LINE Corporation
+ *
+ *  LINE Corporation licenses this file to you under the Apache License,
+ *  version 2.0 (the "License"); you may not use this file except in compliance
+ *  with the License. You may obtain a copy of the License at:
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *  License for the specific language governing permissions and limitations
+ *  under the License.
+ */
+package com.linecorp.armeria.server.docs;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.linecorp.armeria.internal.docs.DocServiceUtil.unifyFilter;
+import static java.util.Objects.requireNonNull;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.BiFunction;
+
+import com.google.common.collect.ArrayListMultimap;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ListMultimap;
+
+import com.linecorp.armeria.common.HttpHeaders;
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.server.ServiceRequestContext;
+
+/**
+ * Builds a new {@link DocService}.
+ */
+public final class DocServiceBuilderImpl implements DocServiceBuilder {
```

Comment:
Do we need to split `DocServiceBuilder` into an interface and an implementation? Do you have any specific use case for this? If not, it'd be nice if we could merge this back into `DocServiceBuilder`.

Reply:
I created the interface in order to inject the `DocServiceBuilder`, so it could be any custom implementation. But maybe it's not needed. I'll review it, and  I'll undo this change.